packages/node-cli/src/commands/project/deploy/DeployAction.js

```diff
@@ -70,17 +75,51 @@ module.exports = class DeployAction extends BaseAction {
 				flags.push(COMMAND.FLAGS.APPLY_CONTENT_PROTECTION);
 			}
 
-			const sdkParams = CommandUtils.extractCommandOptions(params, this._commandMetadata);
+			if (params[COMMAND.FLAGS.PREVIEW]) {
+				try {
+					delete params[COMMAND.FLAGS.PREVIEW];
+					flags = flags.slice((0, 2));
```

Comment:
This line is possibly deleting one of the flags (APPLY_CONTENT_PROTECTION)
Is that what we want? 

If yes, I would rather remove it by specifying it directly. If we add any other flag in the future this could be a source of problems.

Reply:
I did this because I just wanted to delete the no_preview and skip_warning flags that are part of the constant. This only deletes those two. Any other additional flags that are passed, are kept and sent to the preview command in CLI for Java.

Reply:
ok, however I would still specify the removal of the two flags. I'm thinking about maintaining the code in the future. This numbers appear to me like ["magic numbers"](https://en.wikipedia.org/wiki/Magic_number_(programming)) 
Also, using double parenthesis could lead also to misinterpretations (as in the comment of Karl and myself, I didn't know the difference :S ). You could use instead `splice` 

I think it should be something like:
`flags.splice(flags.indexOf(COMMAND.FLAGS.NO_PREVIEW),1);
 flags.splice(flags.indexOf(COMMAND.FLAGS.SKIP_WARNING),1);
  `